### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by mjibson**
Various improvements. See http://speleotrove.com/decimal/daops.html#refrtoix for the naming choice on the ToIntegral functions.

### CODE REVIEW

File: decimal.go [hunk 1]
```diff
@@ -677,37 +677,61 @@ func (d *Decimal) IsZero() bool {
...
-	frac.Negative = neg
-	integ.Negative = neg
+
+	var icoeff *big.Int
+	if integ != nil {
+		icoeff = &integ.Coeff
+		integ.Exponent = 0
+		integ.Negative = neg
+	} else {
+		icoeff = new(big.Int)
```
Comment (knz) [inline decimal.go]: Skip the allocation if frac == nil

File: decimal.go [hunk 2]
```diff
@@ -677,37 +677,61 @@ func (d *Decimal) IsZero() bool {
...
+	}
+
+	if frac != nil {
+		icoeff.QuoRem(&d.Coeff, e, &frac.Coeff)
+		frac.Exponent = d.Exponent
+		frac.Negative = neg
+	} else {
+		icoeff.Quo(&d.Coeff, e)
```
Comment (knz) [inline decimal.go]: Skip the computation if integ == nil

File: decimal.go (hunk 1 above)
Comment (mjibson) [inline decimal.go]: This is the `integ == nil` branch, and we already checked if both integ and frac were nil above, so frac can never be nil in this branch.

File: decimal.go (hunk 2 above)
Comment (mjibson) [inline decimal.go]: Same as above: we are in frac == nil, which means integ must not be nil since they both can't be.

File: decimal.go (hunk 1 above)
Comment (knz) [inline decimal.go]: Sorry I hadn't seen the check at the beginning. Please add the explanatory comment here to repeat the assumption.

Comment (mjibson) [inline decimal.go]: Done.

**Comment by knz** [comment]
thanks!

**Pull request merged**